packages/block-editor/src/components/plain-text/index.native.js

```diff
@@ -16,22 +16,39 @@ import styles from './style.scss';
 export default class PlainText extends Component {
 	constructor() {
 		super( ...arguments );
-		this.isIOS = Platform.OS === 'ios';
+		this.isAndroid = Platform.OS === 'android';
 	}
 
 	componentDidMount() {
 		// if isSelected is true, we should request the focus on this TextInput
 		if ( this._input.isFocused() === false && this.props.isSelected ) {
-			this.focus();
+			if ( this.isAndroid ) {
+				/*
+				 * There seems to be an issue in React Native where the keyboard doesn't show if called shortly after rendering.
+				 * As a common work around this delay is used.
+				 * https://github.com/facebook/react-native/issues/19366#issuecomment-400603928
+				 */
+				this.timeoutID = setTimeout( () => {
+					this._input.focus();
+				}, 100 );
+			} else {
+				this._input.focus();
+			}
 		}
 	}
 
 	componentDidUpdate( prevProps ) {
-		if ( ! this.props.isSelected && prevProps.isSelected && this.isIOS ) {
```

Comment:
When the hide keyboard button is selected componentDidUpdate is called with the changes of isSelected toggling from true to false the is iOS flag here was preventing android from getting the blur call to dismiss the keyboard